### PR TITLE
Withdraw liquidity work from Usama

### PR DESCRIPTION
**Pull request by TBomberman**
- User is able to withdraw liquidity. User will select the amount in percentage from the slider or they can select a static amount in percent given in the buttons below the slider. Then the user will first approve the transaction. Once the transaction is approved the remove button will be enabled which will be used to remove liquidity.
- Install prettier extension from vs code. Press Shift+alt+f to pretty the file.
- Repay Loan design
@Pochetes or @dam417 
Thank you!

### CODE REVIEW

**Review by Pochetes — COMMENTED** [review]
## `react-ranger` is having dependency issues
per this [issue](https://github.com/TanStack/ranger/issues/22), npm finds dep conflicts betwen react and the module. We definitely do not want to downgrade React as it will break everything. Yarn gives a warning but still lets us use it. This is probably good for MVP but we would have to switch in the long run. I say use a different tool but I'd rather let @TBomberman call the shots.

**Comment by Pochetes** [comment]
## Improved `WithdrawLiquidity` Design
Screen Shot 2022-07-16 at 11 36 51 PM

per [commit c0e283d](https://github.com/gammaswap/v1-interface/pull/15/commits/c0e283dbc12629ec3598366d05f1b68f5605d30d)

**Comment by TBomberman** [comment]
Actually I agree with @Pochetes . I think we need an updated react slider and one that is the most popular. Popularity helps with reliability. Are you able to find another one that will fit?

Thank you.

**Comment by UsamaSaleem14** [comment]
react-ranger is a popular slider for react. On yarn it has 23.1k Downloads last 30 days. And we also found its type dependency which removed the error that we were getting

**Comment by TBomberman** [comment]
Ok, but there is still the dependency issue that Roberto mentioned.

**Comment by UsamaSaleem14** [comment]
Yeah I will find a new slider that will be compatible with our react version and also has typescript definition

**Comment by TBomberman** [comment]
Ok, your other PRs are dependent on this as they are merging into Godwin which will then be merged to main.

**Comment by TBomberman** [comment]
Slider has been changed from react-ranger to rc-slider because react-ranger was using an earlier version of React js and we do not want to downgrade our react version.

User is now able to withdraw 100% liquidity. Earlier when user wanted to withdraw 100% liquidity they would get an error. This error occurred because parseFloat would round the number and that would be greater than the actual amount.

Added some style consistency improvement.

thank you @Pochetes

File: components/WithdrawLiquidity/WithdrawLiquidityController.tsx [hunk 1]
```diff
@@ -1,26 +1,269 @@
...
+      let x = y.div(2).add(1)
+      while (x.lt(z)) {
+        z = x
+        x = y.div(x).add(x).div(2)
+      }
+    } else if (!y.isZero()) {
+      z = BigNumber.from(1)
+    }
+    return z
+  }
```
Comment (Pochetes) [inline components/WithdrawLiquidity/WithdrawLiquidityController.tsx:251]: might be better to extract this function into the `utils` folder for reusability.

File: components/WithdrawLiquidity/WithdrawLiquidityController.tsx [hunk 2]
```diff
@@ -1,26 +1,269 @@
...
+        // Deposit poll contract address
+        let address = '0x3eFadc5E507bbdA54dDb4C290cc3058DA8163152'
+        if (accountInfo && accountInfo?.address) {
+          let _depPool = new ethers.Contract(address, DepPool.abi, provider.getSigner(accountInfo?.address))
+          setdepPool(_depPool)
+          const token0Addr = await _depPool.token0()
+          const token1Addr = await _depPool.token1()
+          const _token0 = new ethers.Contract(token0Addr, IERC20Metadata.abi, provider.getSigner(accountInfo?.address))
+          const _token1 = new ethers.Contract(token1Addr, IERC20Metadata.abi, provider.getSigner(accountInfo?.address))
+          const symbol0 = await _token0.symbol()
```
Comment (Pochetes) [inline components/WithdrawLiquidity/WithdrawLiquidityController.tsx:169]: might not be the best way to initialize all of these variables. Separation of concerns is key

**Review by Pochetes — APPROVED** [review]
Remember to replace `*Controller.tsx` files into custom hooks. Shouldn't be difficult to do. Reference this [comment I made](https://github.com/gammaswap/v1-interface/pull/18#discussion_r928494590). The actual components should be in the `pages` folder, while any reusable component should be in the `components` folder. `hooks` for custom hooks specific to a page, and `utils` for very general functions that can be used anywhere. 

Also, `console.log()` statements should be removed before every PR because that is final code that can potentially be merged into `main`.

File: .prettierrc [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+    "printWidth": 200,
+    "tabWidth": 2,
+    "useTabs": false,
```
[the remark this answers is not in the document]
Comment (TBomberman) [reply .prettierrc:4]: Yes preference -> settings -> search 'tab'
image

File: components/WithdrawLiquidity/WithdrawLiquidityController.tsx [hunk 4]
```diff
@@ -1,26 +1,269 @@
...
+    }
+    fetchContract()
+  }, [provider])
+
+  function pretty(num: number) {
+    return ethers.utils
+      .formatEther(num.toString())
+      .toString()
+      ?.match(/^-?\d+(?:\.\d{0,2})?/)![0]
+  }
```
[the remark this answers is not in the document]
Comment (TBomberman) [reply components/WithdrawLiquidity/WithdrawLiquidityController.tsx:197]: This like the numbers formatting function you have in your add liquidity numbers input.

File: components/WithdrawLiquidity/WithdrawLiquidityController.tsx [hunk 5]
```diff
@@ -1,26 +1,269 @@
...
+    } else {
+      amt = ethers.utils.parseEther(((liquidityAmt * balance) / 100).toString()).toString()
+    }
+    if (provider) {
+      // Deposit poll contract address
+      let address = '0x3eFadc5E507bbdA54dDb4C290cc3058DA8163152'
+      if (accountInfo && accountInfo?.address) {
+        setdepPool(new ethers.Contract(address, DepPool.abi, provider.getSigner(accountInfo?.address)))
+      } else {
+        setdepPool(new ethers.Contract(address, DepPool.abi, provider))
```
[the remark this answers is not in the document]
Comment (TBomberman) [reply components/WithdrawLiquidity/WithdrawLiquidityController.tsx:79]: we'll do this going forward and at some point down the line for this case

File: components/WithdrawLiquidity/WithdrawLiquidityController.tsx (hunk 1 above)
Comment (UsamaSaleem14) [reply components/WithdrawLiquidity/WithdrawLiquidityController.tsx:251]: Ok

File: components/WithdrawLiquidity/WithdrawLiquidityController.tsx (hunk 2 above)
Comment (UsamaSaleem14) [reply components/WithdrawLiquidity/WithdrawLiquidityController.tsx]: Fixed it

File: components/WithdrawLiquidity/WithdrawLiquidityController.tsx (hunk 1 above)
Comment (UsamaSaleem14) [reply components/WithdrawLiquidity/WithdrawLiquidityController.tsx]: Done

File: utils/mathFunctions.tsx [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+      x = y.div(x).add(x).div(2)
+    }
+  } else if (!y.isZero()) {
+    z = BigNumber.from(1)
+  }
+  return z
+}
+
+// This function will return an ether value for a wei value provided as a parameter. match will convert the value to two decimal points without rounding the value to nearest big integer
+export function weiToEther(num: number) {
```
Comment (TBomberman) [inline utils/mathFunctions.tsx:20]: this is not needed

File: components/WithdrawLiquidity/WithdrawLiquidityController.tsx [hunk 7]
```diff
@@ -159,27 +159,44 @@ const WithdrawLiquidity = () => {
...
-          const _token1 = new ethers.Contract(token1Addr, IERC20Metadata.abi, provider.getSigner(accountInfo?.address))
-          const symbol0 = await _token0.symbol()
-          const symbol1 = await _token1.symbol()
-          setToken0({address: token0Addr, symbol: symbol0, contract: _token0})
-          setToken1({address: token1Addr, symbol: symbol1, contract: _token1})
-        } else {
-          let _depPool = new ethers.Contract(address, DepPool.abi, provider)
+
+        // Variable to hold deposit pool contract
+        let _depPool = null
```
Comment (TBomberman) [inline components/WithdrawLiquidity/WithdrawLiquidityController.tsx:164]: separate these into a method called initializeTokens()

**Pull request merged**